packages/auth/src/client/service.ts

```diff
@@ -91,56 +88,39 @@ async function verifySig(
   return crypto.verify(null, data, publicKey, Buffer.from(sig))
 }
 
-function validateRequiredJwkProperties(jwk: JWKWithRequired): boolean {
-  if (
-    jwk.kty !== 'OKP' ||
-    (jwk.use && jwk.use !== 'sig') ||
-    (jwk.key_ops &&
-      (!jwk.key_ops.includes('sign') || !jwk.key_ops.includes('verify'))) ||
-    jwk.alg !== 'EdDSA' ||
-    jwk.crv !== 'Ed25519'
-  ) {
-    return false
-  }
-
-  return true
-}
-
-async function validateClientWithRegistry(
+async function validateClient(
   deps: ServiceDependencies,
   clientInfo: ClientInfo
 ): Promise<boolean> {
-  const { config } = deps
+  if (!isClientInfo(clientInfo)) return false
+
   const { jwk } = clientInfo.key
-  const { keyRegistries } = config
 
-  // No key registries in list means no validation
-  if (keyRegistries.length === 0) return true
-  if (!validateRequiredJwkProperties(jwk)) return false
+  const key = await getKeyByKid(deps, jwk.kid)
 
-  const kidUrl = new URL(jwk.kid)
-  if (!keyRegistries.includes(kidUrl.origin)) return false
+  if (!key || !isJWKWithRequired(key) || jwk.x !== key.x || key.revoked)
+    return false
 
-  const { keys, ...registryClientInfo } = await getRegistryDataByKid(
-    deps,
-    jwk.kid
+  if (
+    jwk.client.name !== key.client.name ||
+    jwk.client.uri !== key.client.uri ||
+    jwk.client.id !== key.client.id ||
+    clientInfo.display.name !== key.client.name ||
+    clientInfo.display.uri !== key.client.uri
   )
-
-  if (!keys || !keys[0].kid || !keys[0].x) {
     return false
-  }
 
-  return !!(
-    verifyClientDisplay(clientInfo.display, registryClientInfo) &&
-    verifyJwk(jwk, keys[0])
-  )
+  if (key.exp && new Date() >= new Date(key.exp * 1000)) return false
+  if (key.nbf && new Date() < new Date(key.nbf * 1000)) return false
```

Comment:
The previous validation check (`verifyJwk`) required `key.exp` and `key.nbf`. Is this more correct?

Reply:
Neither `key.exp` or `key.nbf` are required to be defined (according to the type definition). So I shouldn't need to check that they both are. If they are required I can put them in the typeguard and change the typings.

All the tests passed, so I assumed it was working as @njlie intended  🤷

Reply:
The type definition is correct, so this is more accurate to what I intended.